dash/dash-renderer/src/persistence.js

```diff
@@ -306,6 +306,55 @@ const getProps = layout => {
     };
 };
 
+export function setPersistance(layout, newProps, dispatch) {
```

Comment:
@nickmelnikov82 my apologies for not reviewing this quicker!

Can you explain the differences between this new function and `recordUiEdit`? Ideally we'd 🌴 combine them into one (which we could call `recordEdit` I guess) with just a flag to distinguish the two situations, then any differences would be crystal clear.

The differences I see:
- Here you needed to create `finalPersistence` - makes sense because the UI can't edit `persistence` - but in fact I don't see a harm in including this in `recordUiEdit`, it's not a big overhead. Which brings up that in the callback case it's also possible to alter `persistence_type`, so presumably we should also create `finalPersistenceType`.
- Why the short-circuit on `finalPersistence !== persistence`? I guess maybe it's more complicated in this case, at least if you change from one truthy `persistence` value to another, but it seems to me like we'd still want the new value stored.
- Can you explain the `storage.removeItem` case? Is that situation accessible from the UI as well (ie a case we were always missing)?
- `recordUiEdit` has `const vals = originalVal === undefined ? [newVal] : [newVal, originalVal];` - ie we don't store `undefined`. I'm not sure the practical consequences of this, but presumably we want to keep that behavior?

Reply:
1. About finalPersistenceType it makes sense
2. Yes, it's related to case when one truthy persistence value to another. When persistence is changed, the current value is immediately written to the storage as the original value.
3. This is for the case described in the [documentation](https://dash.plotly.com/persistence#explicitly-clearing-saved-data)
4. Without these lines, more than half of the tests fail. I'm not quite sure how storage without the original value is used.